src/input/json_to_storage.cc

```diff
@@ -281,22 +671,22 @@ void JSONToStorage::read_stream(istream &in, const Type::TypeBase &root_type)
  */
 
 
-StorageBase * JSONToStorage::make_storage(JSONPath &p, const Type::TypeBase *type)
+StorageBase * JSONToStorage::make_storage(PathBase *p, const Type::TypeBase *type)
 {
     ASSERT(type != NULL, "Can not dispatch, NULL pointer to TypeBase.\n");
 
     // first check reference
     string ref_address;
-    if (p.get_ref_from_head(ref_address)) {
+    if (p->get_ref_from_head(ref_address)) {
         // todo: mark passed references and check cyclic references
 
         // dereference and take data from there
-        JSONPath ref_path = p.find_ref_node(ref_address);
+    	PathBase * ref_path = p->find_ref_node(ref_address);
         return make_storage( ref_path, type );
     }
```

Comment:
Need to delete ref_path !! Why is it necessary to convert all references (mostly safe) to pointers?